File: Coding Challenges/DailyChallenge/JavaScript/20260110_TicTacToe.js
function ticTacToe(board) {
  for(let index = 0; index < board.length; index++) {
    let row = board[index];
    let col = [board[0][index],board[1][index],board[2][index]];

    if(row.every(isSameSymbol)) {
      return `${row[0]} wins`;
    };
    
    if(col.every(isSameSymbol)) {
      return `${board[0][index]} wins`;
    };
  };

  if([board[0][0], board[1][1], board[2][2]].every(isSameSymbol)
  || [board[0][2], board[1][1], board[2][0]].every(isSameSymbol)
  ) {
    return `${board[1][1]} wins`;
  }

  return 'Draw';
}

function isSameSymbol(item, index, arr) {
  if(index == 0) {  
    return true;
  } 
  else {
    return item == arr[index - 1];
  } 
}

File: Coding Challenges/Daily Challenge/2026/January/20260110-Tic-Tac-Toe.py
def tic_tac_toe(board):
    for index, row in enumerate(board):
        row = board[index]
        col = [board[0][index],board[1][index],board[2][index]]

        if all(item == row[0] for item in row):
            return f'{row[0]} wins'

        
        if all(item == col[0] for item in col):
            return f'{col[0]} wins'

    diag1 = [board[0][0], board[1][1], board[2][2]]
    diag2 = [board[0][2], board[1][1], board[2][0]]
    if (
        all(item == board[1][1] for item in diag1)
        or all(item == board[1][1] for item in diag2)
    ):
        return f'{board[1][1]} wins';
    
    return 'Draw'